File: PB001.fs/Algorithm.fs
namespace ProjectEuler

type public Algorithm() = class
    let run = 
        seq {1..1000}
        |> Seq.filter (fun (x) -> ((x % 3 = 0) || (x % 5 = 0)))
        |> Seq.sum

    interface IAlgorithm with
        member this.Compute(): string = run |> string
        member this.Prepare(): bool = true
    end
end

File: PB001.cs/Algorithm.cs
﻿using System;

namespace ProjectEuler
{
    public class Algorithm : IAlgorithm
    {
        private int run(int n)
        {
            int sum = 0;
            for (int i = 1; i < n; i++)
                if ((i % 3 == 0) || (i % 5 == 0))
                    sum += i;
            return sum;
        }

        public string Compute()
        {
            return string.Format("{0}", run(1000));
        }

        public bool Prepare()
        {
            return true;
        }
    }
}
